### PR TITLE
Feature/webdriverio request interception

### PR DESCRIPTION
**Pull request by szucst**

### CODE REVIEW

File: webDriverIO/specs/data/list-users-schema.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+const userSchema = {
```
Comment (szikszail) [inline webDriverIO/specs/data/list-users-schema.ts]: I'd suggest defining these schemas using the built-in options of the JSON Schema (e.g., definitions, references, etc.).

**Review by szikszail — CHANGES_REQUESTED** [review]

File: webDriverIO/tests/request-validation.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import { expect } from 'chai';
+
+const {
+    expectedURL,
+    expectedResponseStatusCode,
+    expectedRequestMethod,
+    usersSelector
+} = require('../data/test-data.json');
```
Comment (mgabor18) [inline webDriverIO/tests/request-validation.ts]: Maybe you could use import here as well.
(and in the other files too.)

File: webDriverIO/tests/response-modification.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+        it('should return 202 status code after modifying the response', async () => {
+            request = await browser.getRequest(0);
+            request.response.statusCode = modifiedResponseStatusCode;
+            expect(request.response.statusCode).to.equal(modifiedResponseStatusCode);
+        });
+
+        describe('The modified response body', () => {
+            it('should return the appropriate full body schema', async () => {
+                const request = await browser.getRequest(0);
```
Comment (mgabor18) [inline webDriverIO/tests/response-modification.ts]: Redeclaring `request` here is unnecessary as you use a previously declared one throughout your tests.

File: webDriverIO/tests/request-validation.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        });
+
+        it('should be a GET method', async () => {
+            const request = await browser.getRequest(0);
+            expect(request.method).to.equal(expectedRequestMethod);
+        });
+
+        it('should match the URL', async () => {
+            const request = await browser.getRequest(0);
+            expect(request.url).to.contain(`${ expectedURL }`);
```
Comment (mgabor18) [inline webDriverIO/tests/request-validation.ts]: Why do you use a template literal here for `expectedURL`?

**Review by mgabor18 — APPROVED** [review]

**Pull request merged**